### ISSUE TITLE
[Reading] Sigmoid Loss for Language Image Pre-Training

### ISSUE THREAD

**Issue by ShotaSato01** [ShotaSato01/Paper_survey#103]
## 0. Paper
https://arxiv.org/abs/2303.15343

## 1. What is it?
本研究では，Sigmoid loss for Language-Image Pre-training (SigLIP)を提案する．
→標準的なソフトマックス正規化を用いたコントラスト学習とは異なり，シグモイド損失は画像とテキストのペアのみに基づいて動作し，正規化のためにペア間類似度のグローバルな情報を必要としない

既存のコントラスト目的関数は，バッチレベルのsoftmaxベースのコントラスト損失によって実現される
→softmaxの単純な実装では，数値的に不安定であるため，softmaxの前に正規化することで安定化をはかるが，これはバッチ全体をもう一度走査する必要がある

本論文では，シグモイド損失を代替にすることを提案する．
→これによって，バッチ全体に跨る演算を必要とせず，分散環境での損失実装を大幅に単純化し，効率を向上させる．
SigLIP：CLIPの損失関数にシグモイド型損失を利用する
SigLiT：LiTの損失関数にシグモイド型損失を利用する

<img width="622" height="664" alt="Image" src="https://github.com/user-attachments/assets/507214c4-6464-44b4-8e30-4416f92f1abc" />

## 3. Where is the key to technologies and techniques?
**Softmax loss for language image pre-training**
画像モデル$`f(・)`$とテキストモデル$`g(・)`$は次の目的関数を最小化するように学習される

<img width="1035" height="297" alt="Image" src="https://github.com/user-attachments/assets/ab536fd3-a85f-4c61-9328-0d762d130d38" />
  
→$`x_{i}, y_{i}`$はそれぞれ，ViT，Transformerアーキテクチャを採用する
　→softmax損失は非対称であるため，正規化は「画像方向」と「テキスト方向」で独立に2回行われる

**Sigmoid loss for language image pre-training**
softmax損失の代替として，グローバルな正規化項を必要としない，より単純な代替手法を提案する

シグモイドベースの損失
→各画像-テキストのペアを独立処理する(バッチに依存しない)
　→全てのペアの組み合わせからなるデータセットに対する標準的な二値分類問題へと学習問題を変換することに相当

<img width="709" height="221" alt="Image" src="https://github.com/user-attachments/assets/5d30074c-a0e5-4d7e-a49b-902fa5d33bdb" />
  
→初期状態では，負例が多数存在することによる強い不均衡のため，損失が負例に支配され，初期の最適化ステップが大きくなりすぎる
　→温度パラメータtと同様に追加の学習可能バイアス項bを導入する

**Efficient “chunked” implementation**

<img width="1298" height="677" alt="Image" src="https://github.com/user-attachments/assets/011055ee-45ee-4457-960d-b89ea4717970" />
  
→シグモイド損失では各ペアが独立項であるため，効率的な損失の計算が実現できる

## 4. How did evaluate it?
**各モデルでのバッチサイズごとの性能比較**

<img width="1144" height="625" alt="Image" src="https://github.com/user-attachments/assets/203e4b34-e2fb-4d59-af92-89e23937a300" />

<img width="619" height="605" alt="Image" src="https://github.com/user-attachments/assets/f02343c7-55c6-420b-890c-4bb8da7f7a92" />
  
→SigLiTにおけるバッチサイズごと/損失関数ごとの性能の比較

<img width="591" height="514" alt="Image" src="https://github.com/user-attachments/assets/41e5a149-76b4-4975-9f9d-b46649e7bb35" />
  
→mSigLIPの言語ごとの結果

<img width="527" height="661" alt="Image" src="https://github.com/user-attachments/assets/31472591-7d41-4bf6-b202-954814b374b5" />
  
→事前学習済みの重みを使ってSigLIPを効率的に学習する方法についての検討
　事前学習ずみ重みにWeight Decayを適用しない設定が最良

<img width="932" height="722" alt="Image" src="https://github.com/user-attachments/assets/a5dfb083-e0b0-40fb-aec0-36fbeee1cb08" />
  
→モデルをスケーリングさせた時の結果

<img width="1027" height="539" alt="Image" src="https://github.com/user-attachments/assets/213c0af0-44ba-4837-bcf2-e7623f90e8d4" />
  
→負例の比率による比較
・Random：ランダムに負例を除外
・Hard：最も損失が高い負例を保持
・Easy：最も損失が低い負例を保持
・Hard, Matched pair：学数ステップ増加で総ペア数を帰る

<img width="858" height="534" alt="Image" src="https://github.com/user-attachments/assets/871c9e7d-074f-42f4-915d-bebf07b7c613" />
  
→バイアス項の比較

<img width="1648" height="546" alt="Image" src="https://github.com/user-attachments/assets/b7c7ed21-8a9b-4d58-bc89-f2c716b5ef80" />
  
→ラベルノイズいに対する頑健性評価

## 5. Is there a discussion?
比較的控えめな32kのバッチサイズで最適な性能が得られる，という予想外の結論に至った
